### DISCUSSION TITLE
Proper usage of `NetworkBase::isConnected()`

### DISCUSSION THREAD

**Discussion by paliasgh** [robotology/community#680] [Q&A] [answered]
Hello,

I was trying to use `yarp::os::NetworkBase::isConnected()` method to check in my code if two ports are connected, but I could not obtain a desirable behaviour. I have this line in my C++ code:

`yarp::os::Network::isConnected("/icub-pli/tagsRight:o","/icub-pli/tagsRight:i",false); `

When the two ports are not connected, it prints:

`[INFO] |yarp.os.Network| No connection from /icub-pli/tagsRight:o to /icub-pli/tagsRight:i found `

Once I connect the ports through the `yarpmanager`, it prints these two lines, but still returns false:

```
[INFO] |yarp.os.Network| Connection found between /icub-pli/tagsRight:o and /icub-pli/tagsRight:i using carrier fast_tcp 
[INFO] |yarp.os.Network| No connection from /icub-pli/tagsRight:o to /icub-pli/tagsRight:i found 
```

Could you please help me with that? Why does the function still return false? Thanks!

**Comment by Nicogene** [comment, marked as answer]
Hi @paliasgh,

it is a while since I don't work on YARP so my memories could be wrong, but if I recall correctly that `isConnected` you invoked checks if there is a connection between the `src` and `dst` using the default carrier, i.e. `tcp`.

Your ports are connected using `fast_tcp`, then you should use [this function](https://www.yarp.it/latest/classyarp_1_1os_1_1NetworkBase.html#a2e687b9923440368017c16be9d1f36cf) as follows:

```c++
yarp::os::Network::isConnected("/icub-pli/tagsRight:o","/icub-pli/tagsRight:i", "fast_tcp",false);
```
> it is a while that I don't work on YARP so my memories could be wrong, but if I recall correctly that `isConnected` you invocked checks if there is a connection between the `src` and `dst` using the default carrier, i.e. `tcp`.

Unfortunately, this is not made explicit in the documentation.

cc @traversaro @randaz81

**Reply by traversaro** [reply]
I tried to fix this in https://github.com/robotology/yarp/pull/3075 .
